plugins/CustomPiwikJs/TrackingCode/PluginTrackerFiles.php

```diff
@@ -71,9 +71,32 @@ public function find()
             }
         }
 
+        foreach ($jsFiles as $plugin => $file) {
+            if (!$this->shouldIncludeFile($plugin)) {
+                unset($jsFiles[$plugin]);
+            }
+        }
+
         return $jsFiles;
     }
 
+    protected function shouldIncludeFile($pluginName)
+    {
+        $shouldAddFile = true;
+
+        /**
+         * Detect if a custom tracker file should be added to the piwik.js tracker or not.
+         *
+         * This is useful for example if a plugin only wants to add its tracker file when the plugin is configured.
+         *
+         * @param bool &$shouldAddFile Decides whether the tracker file belonging to the given plugin should be added or not.
+         * @param string $pluginName The name of the plugin this file belongs to
+         */
+        Piwik::postEvent('CustomPiwikJs.shouldAddTrackerFile', array(&$shouldAddFile, $pluginName));
```

Comment:
FYI: I was also thinking about posting the event here: https://github.com/piwik/piwik/blob/3.0.4/plugins/CustomPiwikJs/TrackingCode/PiwikJsManipulator.php#L33 but then decided to post it here instead. This is in case the list of files will be also used for something else. Eg if there was a command to list which files were included in piwik.js etc. Can also change it if needed